modules/storage/config.go

```diff
@@ -47,8 +48,8 @@ func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet)
 	cfg.Trace.Block.SearchPageSizeBytes = 1024 * 1024 // 1 MB
 
 	cfg.Trace.Azure = &azure.Config{}
-	f.StringVar(&cfg.Trace.Azure.StorageAccountName.Value, util.PrefixConfig(prefix, "trace.azure.storage-account-name"), "", "Azure storage account name.")
-	f.StringVar(&cfg.Trace.Azure.StorageAccountKey.Value, util.PrefixConfig(prefix, "trace.azure.storage-account-key"), "", "Azure storage access key.")
+	f.StringVar(&cfg.Trace.Azure.StorageAccountName.Value, util.PrefixConfig(prefix, "trace.azure.storage-account-name"), os.Getenv("AZURE_STORAGE_ACCOUNT_NAME"), "Azure storage account name.")
```

Comment:
I'll look again because that would be my preference also.  Doing it here means that the current client implementation can remain the same.

https://github.com/grafana/tempo/blob/main/tempodb/backend/azure/azure_helpers.go#L23

I also looked briefly at updating the client, since the one we are using seems to be deprecated, or at least they are encouraging users to migrate.  But I didn't have the context on all the changes in that update since I'm new to Azure and learning.

Reply:
If they are loading environment variables in the client, I don't see where.  And they don't make it easy to find if they support some.  I think the updated client might though.